# HG changeset patch
# User Zhigang Wang <zhigang.x.wang@oracle.com>
# Date 1310145146 -3600
# Node ID 443c6a7b6079f490d9adbb46352c20c2628fb144
# Parent  0648846b4d17c5b09c11ffe48a49fbb7616f630e
tools: fix shared loop device detection in block hotplug script

We need to cope with 0 or more '0' prefixed to '$dev':

  # stat -c '%D' "/system.img"
  13
  # losetup -a
  /dev/loop0: [0013]:12062248 (/system.img)

Signed-off-by: Zhigang Wang <zhigang.x.wang@oracle.com>
Committed-by: Ian Jackson <ian.jackson@eu.citrix.com>

Backported to xen-4.1.0 by: Marek Marek Marczykowski <marmarek@mimuw.edu.pl>

--- xen-4.1.0/tools/hotplug/Linux/block.orig	2011-07-24 15:20:04.000000000 +0200
+++ xen-4.1.0/tools/hotplug/Linux/block	2011-07-24 15:20:53.541126891 +0200
@@ -285,8 +285,8 @@
           fi
 
           shared_list=$(losetup -a |
-                sed -n -e "s@^\([^:]\+\)\(:[[:blank:]]\[${dev}\]:${inode}[[:blank:]](${file})\)@\1@p" )
-          for dev in "$shared_list"
+                sed -n -e "s@^\([^:]\+\)\(:[[:blank:]]\[0*${dev}\]:${inode}[[:blank:]](.*)\)@\1@p" )
+          for dev in $shared_list
           do
             if [ -n "$dev" ]
             then
